### PR TITLE
Remove difficulty spike nerf from the Flashlight skill in osu!

### PR DESCRIPTION
**Pull request by apollo-dw**
`OsuStrainSkill`, by default, reduces the highest strains in the final list of strains for a map. The idea behind doing so is that the highest strains will decrease their influence on the overall difficulty of the map. The reason this doesn't make sense for the flashlight skill is because it is memory-based instead of mechanics-based. For example, having only a few "hard-to-memorise" patterns in a map doesn't make it trivial to memorise - the difficulty to memorise those patterns still exists in it's fullest.

All in all, the values shouldn't vary by a whole lot by this, but this is in theory more accurate.

The legendary discussion that took place was me spotting that Flashlight nerfs difficulty spikes, and me asking MBMasher (who developed the Flashlight skill) whether they also thought it made no sense. [This can be seen here, in the pp dev discord.](https://discord.com/channels/546120878908506119/549012787733594115/980897976132190228)

---

SR/PP changes spreadsheet: https://docs.google.com/spreadsheets/d/1RGtONEOfAgBE6WX-htrJTzxr5w_B2aCqZA8eM7iGL6o/edit?usp=sharing

As of https://github.com/ppy/osu/pull/18791/commits/630bd244d5af115226bac1a25d3e245c193ea571

### CODE REVIEW

**Comment by peppy** [comment]
Please fill out the OP with a description of what this change does / why / any discussions which led to it. As it stands I have no idea what I'm looking at.

File: osu.Game.Rulesets.Osu/Difficulty/Skills/Flashlight.cs [hunk 1]
```diff
@@ -28,6 +28,7 @@ public Flashlight(Mod[] mods)
         private double skillMultiplier => 0.05;
         private double strainDecayBase => 0.15;
         protected override double DecayWeight => 1.0;
+        protected override int ReducedSectionCount => -1;
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Difficulty/Skills/Flashlight.cs:31]: On the code side -1 is really weird as well as 0 should work the same and actually makes semantic sense.

**Comment by apollo-dw** [comment]
`DifficultyMultiplier` is one thing, but that can be easily re-added with a comment explaining why it exists. Does that sound fine?

**Review by stanriders — DISMISSED** [review]

**Comment by smoogipoo** [comment]
SR/PP changes sheet in PR description.

**Review by MBmasher — DISMISSED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**